_release/bat/storage_bat/storage_bat_test.go

```diff
@@ -634,3 +635,88 @@ func TestAttachToStoppedInstance(t *testing.T) {
 			volumeID, instanceID, err)
 	}
 }
+
+// Check that volumes created from image work.
+//
+// Create a volume from an image and check that the sizes match.
+//
+// The created volume should be the same size as the image to the nearest GiB
+// rounded up.
+func TestCreateVolumeFromImage(t *testing.T) {
+	ctx, cancelFunc := context.WithTimeout(context.Background(), standardTimeout)
+	defer cancelFunc()
+
+	images, err := bat.GetImages(ctx, "")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var imageUUID string
+	var image *bat.Image
+	for imageUUID, image = range images {
+		break
+	}
+
+	volumeUUID, err := bat.AddVolume(ctx, "", imageUUID, "image", &bat.VolumeOptions{})
```

Comment:
If there are no images this will probably fail which would be a bit misleading.  Might be best to skip the rest of the test if imageUUID == "" or len(images) == 0